### PR TITLE
feat: display inscriptions from native segwit address

### LINKED ISSUE #4018: Show inscriptions on Native SegWit address in collectibles

### PR DESCRIPTION
**Pull request by alter-eggo**
> Try out this version of the Leather — download [extension builds](https://github.com/leather-wallet/extension/actions/runs/6196508994).

See title, closes #4018

### CODE REVIEW

File: src/app/query/bitcoin/ordinals/use-inscriptions.query.ts [hunk 1]
```diff
@@ -93,11 +93,13 @@ export function useTaprootInscriptionsInfiniteQuery() {
...
       while (addressesWithoutOrdinalsNum < stopSearchAfterNumberAddressesWithoutOrdinals) {
-        const response = await fetchInscriptions(
-          Object.keys(addressesData),
-          offset,
-          inscriptionsLazyLoadLimit
-        );
+        const addresses = Object.keys(addressesData);
+
+        // add native segwit address to 1 chunk of addresses
+        if (fromIndex === 0) {
```
Comment (alter-eggo) [inline src/app/query/bitcoin/ordinals/use-inscriptions.query.ts:99]: don't much like this, but couldn't find good solution now, will try to find better place to insert native segwit address

File: src/app/features/collectibles/components/bitcoin/hooks/use-ordinals-render.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import { useEffect } from 'react';
+import { useInView } from 'react-intersection-observer';
+
+export function useOrdinalsRender({ fetchNextPage }: { fetchNextPage: () => Promise<void> }) {
```
Comment (kyranjamie) [inline src/app/features/collectibles/components/bitcoin/hooks/use-ordinals-render.ts]: This is more a generic fetch next page fn vs something ordinals specific?

File: src/app/query/bitcoin/ordinals/use-inscriptions.query.ts (hunk 1 above)
Comment (kyranjamie) [reply src/app/query/bitcoin/ordinals/use-inscriptions.query.ts:99]: Not sure what this actually does? 🤔

Comment (alter-eggo) [reply src/app/query/bitcoin/ordinals/use-inscriptions.query.ts:99]: here I just add ns address to 1 chunk of taproot addresses

File: src/app/features/collectibles/components/bitcoin/hooks/use-ordinals-render.ts (hunk 2 above)
Comment (alter-eggo) [reply src/app/features/collectibles/components/bitcoin/hooks/use-ordinals-render.ts]: here it assumes ordinals specific

Comment (kyranjamie) [reply src/app/features/collectibles/components/bitcoin/hooks/use-ordinals-render.ts]: Where is the ordinals specific code?

Comment (alter-eggo) [reply src/app/features/collectibles/components/bitcoin/hooks/use-ordinals-render.ts]: decided to remove it, seems no need in this hook

**Review by fbwoolf — APPROVED** [review]

**Review by markmhendrickson — APPROVED** [review]
My inscription on the Native SegWit address shows up

**Review by kyranjamie — APPROVED** [review]
To merge?

**Comment by alter-eggo** [comment]
better to hold it, until we implement support for sending inscriptions from native segwit address

**Comment by markmhendrickson** [comment]
@alter-eggo should we QA this PR now with the incorporation of https://github.com/leather-wallet/extension/pull/4225 to ensure that inscriptions show on Native SegWit addresses and can be sent from there?

**Comment by markmhendrickson** [comment]
@fabric-8 to QA and @markmhendrickson to provide testing instructions

**Comment by markmhendrickson** [comment]
Testing instructions:

1. Deposit or mint new Ordinal inscription using the Bitcoin address that appears under "Tokens" in [the receive modal](https://github.com/leather-wallet/extension/assets/28991/5bd33536-8f75-4fbd-af34-bf4be3b30103). It will start with `bc1q`. You can send an existing inscription to that address from your wallet under "Collectibles" or use [Gamma's inscription tool](https://gamma.io/?is_create_open=true&path=inscription) and provide this same address when asked for your inscription deposit address.
2. Wait for the send or mint to confirm on chain
3. Refresh the collectibles on the home screen using [the 🔄 option](https://github.com/leather-wallet/extension/assets/28991/1160393a-2d8d-4b6b-b5ef-527431823bb4)
4. Confirm that the newly sent or minted inscription shows up under "Collectibles"
5. Select "Send" for this inscription and send it to another address (it doesn't matter where, but you can choose a BTC address for another one of your accounts in the wallet)
6. Wait for the send to confirm on chain
7. Refresh the collectibles on the home screen using [the 🔄 option](https://github.com/leather-wallet/extension/assets/28991/1160393a-2d8d-4b6b-b5ef-527431823bb4)
8. Confirm that the inscription no longer appears under "Collectibles" for this account but does appear in the recipient account

**Comment by alter-eggo** [comment]
@markmhendrickson yeah, that would be great

**Comment by fabric-8** [comment]
Tested this and looks like it's working as expected! I went from 

1. _Account 1:_ Native SegWit → Taproot
2. _Account 1:_ Taproot → _Account 2:_ Native SegWit

Ordinal appeared under "collectibles" each time as expected

**Review by fabric-8 — APPROVED** [review]

**Pull request merged**